ctest-next/src/generator.rs

```diff
@@ -65,20 +105,489 @@ impl TestGenerator {
     ///
     /// This is useful for if the C library is installed to a nonstandard
     /// location to ensure that compiling the C file succeeds.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use std::env;
+    /// use std::path::PathBuf;
+    ///
+    /// use ctest_next::TestGenerator;
+    ///
+    /// let mut cfg = TestGenerator::new();
+    /// let out_dir = PathBuf::from(env::var_os("OUT_DIR").unwrap());
+    /// cfg.include(out_dir.join("include"));
+    /// ```
     pub fn include<P: AsRef<Path>>(&mut self, p: P) -> &mut Self {
         self.includes.push(p.as_ref().to_owned());
         self
     }
 
     /// Configures the output directory of the generated Rust and C code.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use ctest_next::TestGenerator;
+    ///
+    /// let mut cfg = TestGenerator::new();
+    /// cfg.out_dir("path/to/output");
+    /// ```
     pub fn out_dir<P: AsRef<Path>>(&mut self, p: P) -> &mut Self {
         self.out_dir = Some(p.as_ref().to_owned());
         self
     }
 
+    /// Skipped item names are printed to `stderr` if `skip` is `true`.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use ctest_next::TestGenerator;
+    ///
+    /// let mut cfg = TestGenerator::new();
+    /// cfg.verbose_skip(true);
+    /// ```
+    pub fn verbose_skip(&mut self, skip: bool) -> &mut Self {
+        self.verbose_skip = skip;
+        self
+    }
+
+    /// Is field volatile?
+    ///
+    /// This is used to make sure that the generated test code also has the volatile keyword.
```

Comment:
```suggestion
    /// Indicate that a struct field should be marked `volatile`.
```

Complete sentence for the summary, then I don't think we need the rest